lib_trycatch/src/trycatch_impl.h

```diff
@@ -19,11 +19,17 @@ typedef struct typecatch_try_t {
 #define ADDRESS_OF &
 #endif
 
+#if defined(__cplusplus) || defined(__XC__)
```

Comment:
Hi @samchesney, you're right. It's only needed when using the library from C++. I got confused since for lib_mp3, the functions that needed to be declared as extern C were C++ functions that were being called from both xc and c++ files, so needed extern C for both c++ and xc. Since lib_trycatch is implemented in C, extern C for xc is not needed. I'll change this and update the PR. Thanks!